File: app/src/main/java/io/bos/accountmanager/core/utils/SPUtil.java
package io.bos.accountmanager.core.utils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 *
 * Created by Administrator on 2018/12/21/021.
 */

public class SPUtil {
    private final String SP_NAME = "language_setting";
    private final String TAG_LANGUAGE = "language_select";
    private static volatile SPUtil instance;

    private final SharedPreferences mSharedPreferences;

    public SPUtil(Context context) {
        mSharedPreferences = context.getSharedPreferences(SP_NAME, Context.MODE_PRIVATE);
    }


    public void saveLanguage(int select) {
        SharedPreferences.Editor edit = mSharedPreferences.edit();
        edit.putInt(TAG_LANGUAGE, select);
        edit.commit();
    }

    public int getSelectLanguage() {
        return mSharedPreferences.getInt(TAG_LANGUAGE, 0);
    }

    public static SPUtil getInstance(Context context) {
        if (instance == null) {
            synchronized (SPUtil.class) {
                if (instance == null) {
                    instance = new SPUtil(context);
                }
            }
        }
        return instance;
    }

}

File: app/src/main/java/io/bos/accountmanager/utils/SPUtil.kt
package io.bos.accountmanager.utils

import android.content.Context
import android.content.SharedPreferences

import java.util.Locale

/**
 * Created by Administrator on 2019/1/9/009.
 */

class SPUtil(context: Context) {
    private val SP_NAME = "language_setting"
    private val TAG_LANGUAGE = "language_select"
    private val TAG_SYSTEM_LANGUAGE = "system_language"

    private val mSharedPreferences: SharedPreferences

    var systemCurrentLocal = Locale.ENGLISH

    val selectLanguage: Int
        get() = mSharedPreferences.getInt(TAG_LANGUAGE, 0)


    init {
        mSharedPreferences = context.getSharedPreferences(SP_NAME, Context.MODE_PRIVATE)
    }


    fun saveLanguage(select: Int) {
        val edit = mSharedPreferences.edit()
        edit.putInt(TAG_LANGUAGE, select)
        edit.commit()
    }

    companion object {
        @Volatile private var instance: SPUtil? = null

        fun getInstance(context: Context): SPUtil? {
            if (instance == null) {
                synchronized(SPUtil::class.java) {
                    if (instance == null) {
                        instance = SPUtil(context)
                    }
                }
            }
            return instance
        }
    }
}
